mne/io/nirx/nirx.py

```diff
@@ -0,0 +1,204 @@
+# Authors: Robert Luke <mail@robertluke.net>
+#
+# License: BSD (3-clause)
+
+import configparser as cp
+import re as re
+import glob as glob
+import pandas as pd
+import numpy as np
+
+from ..base import BaseRaw
+from ..meas_info import create_info
+from ...utils import logger, verbose, fill_doc
+
+
+@fill_doc
+def read_raw_nirx(fname, preload=False, verbose=None):
+    """Reader for a NIRX fNIRS recording.
+
+    Parameters
+    ----------
+    fname : str
+        Path to the NIRX data folder.
+    %(preload)s
+    %(verbose)s
+
+    Returns
+    -------
+    raw : instance of RawNIRX
+        A Raw object containing NIRX data.
+
+    See Also
+    --------
+    mne.io.Raw : Documentation of attribute and methods.
+    """
+    return RawNIRX(fname, preload, verbose)
+
+
+@fill_doc
+class RawNIRX(BaseRaw):
+    """Raw object from a NIRX fNIRS file.
+
+    Parameters
+    ----------
+    fname : str
+        Path to the NIRX data folder.
+    %(preload)s
+    %(verbose)s
+
+    See Also
+    --------
+    mne.io.Raw : Documentation of attribute and methods.
+    """
+
+    @verbose
+    def __init__(self, fname, preload=False, verbose=None):
+        logger.info('Loading %s' % fname)
+
+        # Check if required files exist
+
+        file_dat = glob.glob(fname + '/*.dat')
+        file_evt = glob.glob(fname + '/*.evt')
+        file_hdr = glob.glob(fname + '/*.hdr')
+        file_inf = glob.glob(fname + '/*.inf')
+        file_set = glob.glob(fname + '/*.set')
+        file_tpl = glob.glob(fname + '/*.tpl')
+        file_wl1 = glob.glob(fname + '/*.wl1')
+        file_wl2 = glob.glob(fname + '/*.wl2')
+        file_cfg = glob.glob(fname + '/*config.txt')
+        file_mat = glob.glob(fname + '/*probeInfo.mat')
+
+        assert_one(file_dat, "Should be one dat file")
+        assert_one(file_evt, "Should be one evt file")
+        assert_one(file_hdr, "Should be one hdr file")
+        assert_one(file_inf, "Should be one inf file")
+        assert_one(file_set, "Should be one set file")
+        assert_one(file_tpl, "Should be one tpl file")
+        assert_one(file_wl1, "Should be one wl1 file")
+        assert_one(file_wl2, "Should be one wl2 file")
+        assert_one(file_cfg, "Should be one config file")
+        assert_one(file_mat, "Should be one mat file")
```

Comment:
DRY alarms are going off in my head here. How about:
```
keys = ('dat', 'evt', ...)
files = dict()
for key in keys:
    files[key] = glob.glob('%s/*.%s' % (fname, key))
    if len(files[key]) != 1:
        raise ...
    files[key] = files[key][0]
```
    
files = {key: glob.glob('%s/*.%s' % (fname, key) for key in keys}

for key in keys